models/activities/notification.go

```diff
@@ -459,6 +459,22 @@ func (n *Notification) HTMLURL() string {
 	return ""
 }
 
+// Link formats a URL-string to the notification
+func (n *Notification) Link() string {
+	switch n.Source {
+	case NotificationSourceIssue, NotificationSourcePullRequest:
+		if n.Comment != nil {
```

Comment:
Okay, interesting.
1. I would suggest something like `if n.CommentID > 0`, followed by a `n.LoadComment()`
2. I just noticed that `n.LoadComment()` hasn't been implemented at all for `Notification`, only for `NotificationList`. This is a huge architectural mistake that we will have to correct at some point. 
3. Sounds like a lot of refactoring PRs are still waiting to happen. Let's see when I'll be able to tackle at least some of these problems.

Reply:
I think you are right. But this is just a copy of `HTMLURL` with slight changes. So both functions should be changed in future PRs. I will keep this PR doing its job.